src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -285,7 +301,9 @@ trait Qualif {
     }
 }
 
-// Constant containing interior mutability (UnsafeCell).
+/// Constant containing interior mutability (UnsafeCell).
+/// This must be ruled out to make sure that evaluating the constant at compile-time
+/// and run-time would produce the same result.
```

Comment:
Nope, that's irrelevant. The problem is promoting e.g. `&AtomicUsize` and *then* mutating it.
That is, uses of the promoted references must not be able to distinguish between the promoted and unpromoted case (ignoring pointer addresses).

Reply:
Well that's what I meant.\^\^

And also the part where any use of a `const` item must be equivalent to inlining its definition.

Reply:
I have expanded the description a bit, is this better?